### DISCUSSION TITLE
Difference between standard mixed formulation and solving them as independent function spaces

### DISCUSSION THREAD

**Discussion by bbbyyfff** [multiphenics/multiphenicsx#24] [General]
Hi all, 

I'm going through the tutorials again and I have the following two questions.

1. I am wondering if there is any difference between the mixed formulation and treating them as independent function spaces (as shown in Tutorial 2: stokes problem). 
2. In the problem class defined to interact with SNES, there's a `P_mat` defined. I was wondering if it means some preconditioner, and if there is any examples on using this (or any recommendations on preconditioning?.

Thank you!

**Comment by francesco-ballarin** [comment]
On 1, from the mathematical point of view there isn't any difference. From a programming point of view, personally I like the block formulation (rather than the mixed element one) because I find some aspects of the mixed element implementation odd (e.g., having to use split, or having to collapse subspaces).

On 2, indeed P_mat is the preconditioner. That is not my field so there aren't examples about that in multiphenicsx. However, if you find an example about that in dolfinx it will likely be very similar to what you have to do with multiphenicsx.

**Reply by bbbyyfff** [reply]
Thank you!
